ql/time/daycounters/yearfractiontodate.cpp

```diff
@@ -0,0 +1,71 @@
+/* -*- mode: c++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*
+ Copyright (C) 2023 Klaus Spanderen
+
+ This file is part of QuantLib, a free-software/open-source library
+ for financial quantitative analysts and developers - http://quantlib.org/
+
+ QuantLib is free software: you can redistribute it and/or modify it
+ under the terms of the QuantLib license.  You should have received a
+ copy of the license along with this program; if not, please email
+ <quantlib-dev@lists.sf.net>. The license is also available online at
+ <http://quantlib.org/license.shtml>.
+
+ This program is distributed in the hope that it will be useful, but WITHOUT
+ ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ FOR A PARTICULAR PURPOSE.  See the license for more details.
+*/
+
+#include "yearfractiontodate.hpp"
+
+#include <ql/math/comparison.hpp>
+
+#include <boost/numeric/conversion/cast.hpp>
+#include <cmath>
+
+namespace QuantLib {
+
+    YearFractionToDate::YearFractionToDate(
```

Comment:
Since the constructor can't precompute anything anyway, I'd write this as a function instead of a class so it's easier to call it—I'm seeing a lot of `YearFractionToDate(dc, d)(t)` that could be `yearFractionToDate(dc, d, t)`.

Reply:
done

Reply:
Thanks!